src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,92 @@
-/* eslint-disable */
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import * as todosActions from '../../features/todos';
+import { currentTodoActions } from '../../features/currentTodo';
+import { filterTodos } from '../../app/function';
 
 export const TodoList: React.FC = () => {
-  return (
-    <>
-      <p className="notification is-warning">
-        There are no todos matching current filter criteria
-      </p>
-
-      <table className="table is-narrow is-fullwidth">
-        <thead>
-          <tr>
-            <th>#</th>
-
-            <th>
-              <span className="icon">
-                <i className="fas fa-check" />
-              </span>
-            </th>
-
-            <th>Title</th>
-            <th> </th>
-          </tr>
-        </thead>
-
-        <tbody>
-          <tr data-cy="todo">
-            <td className="is-vcentered">1</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">delectus aut autem</p>
-            </td>
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo) || null;
+  const { todos } = useAppSelector(state => state.todos);
+  const { status } = useAppSelector(state => state.filter);
+  const { query } = useAppSelector(state => state.filter);
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">2</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                quis ut nam facilis et officia qui
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  useEffect(() => {
+    dispatch(todosActions.init());
```

Comment:
Ensure that the `init` action in `todosActions` is correctly defined to initialize the todos state. This is crucial for the component to display the correct list of todos.